tests/usm/usm_api.h

```diff
@@ -0,0 +1,688 @@
+          verifier.call(parent, ptr);
+        } else {
+          static_assert(std::is_same_v<typename caller::type, sycl::queue>,
+                        "Tests with events are not acceptible for handler");
+          // List of events that tested usm operation should wait
+          events_t depEvents(numEvents);
+          // Run time-consuming tasks to generate events
+          for (size_t i = 0; i < depEvents.size(); ++i) {
+            depEvents[i] = gens[i].init(parent, init_value);
+          }
+
+          if constexpr (numEvents == 1) {
+            auto event = verifier.call(parent, ptr, depEvents[0]);
+            event.wait();
+          } else {
+            auto event = verifier.call(parent, ptr, depEvents);
+            event.wait();
+          }
+
+          // Fast copy arrays for future check. If tested function does not
+          // wait for the passed events to complete, then an incompletely
+          // initialized 'src_arr' will be copied to 'dst_arr' and the test
+          // will fail during validation. Reverse order is used to increase
+          // probability of data race.
+          for (size_t i = numEvents; i > 0; --i) {
+            const size_t idx = i - 1;
+            gens[idx].copy_arrays(parent);
+          }
+        }
+      };
+
+      caller::submit(queue, action);
+
+      log.debug("... wait for all actions submitted");
+      queue.wait_and_throw();  // avoid data race
+
+      if constexpr (alloc != allocation::device) {
+        // we can access the pointer from the host side
+        log.debug("... verify USM operation results");
+        verifier.check(ptr);
+      } else {
+        log.debug("... retrieve USM operation results from device allocation");
+        auto values = storage_t::copy_values(queue, scopedStorage);
+
+        log.debug("... verify USM operation results");
+        verifier.check(values.data());
+      }
+
+      bool events_result = true;
+      for (size_t i = 0; i < numEvents; ++i)
+        events_result = events_result && gens[i].check(init_value);
+
+      if (!events_result)
+        FAIL(log, "One or more generators completed work before the verifier."
+                  "Detected mismatch in buffers.");
+
+      log.debug("... destruct the queue");
+    } catch (sycl_cts::util::fail_check &ex) {
+      std::string message{verifier_t::template description<alloc>()};
+      message += " failed for " + type_name_string<T>::get(typeName);
+      message += ". ";
+      message += ex.what();
+      FAIL(log, message);
+    } catch (...) {
+      std::string message{verifier_t::template description<alloc>()};
+      message += " got an exception for " + type_name_string<T>::get(typeName);
+      log.note(message);
+      throw;
+    }
+  }
+};
+
+/** @brief Functor used as the entry point to run every test
+ *  @tparam T Underlying data type for USM or non-USM pointer
+ *  @tparam checkT Defines the actual method to check and the specific test to
```

Comment:
```suggestion
 *  @tparam checkT Defines the actual member function to check and the specific test to
```

Reply:
I've replaced "method" to "member function" in this commit: 800816219c8f8b2a87887df4ef54c8088b0a153b